oak/server/rust/oak_runtime/src/node/mod.rs

```diff
@@ -117,47 +107,50 @@ pub fn check_port(address: &SocketAddr) -> Result<(), ConfigurationError> {
 }
 
 impl Configuration {
-    /// Creates a new Node instance corresponding to the [`Configuration`] `self`.
+    /// Creates a new Node instance corresponding to the [`Configuration`].
     ///
-    /// On success returns a boxed [`Node`] that may be started by invoking the [`Node::start`]
-    /// method.
+    /// On success returns a boxed [`Node`] that can be run with [`Node::run`].
     pub fn create_node(
         &self,
-        config_name: &str, // Used for pretty debugging
-        runtime: RuntimeProxy,
+        node_name: &str, // Used for pretty debugging
+        config_name: &str,
         entrypoint: String,
-        initial_handle: oak_abi::Handle,
-    ) -> Box<dyn Node> {
+    ) -> Option<Box<dyn Node + Send>> {
         debug!(
-            "{:?}: create_node('{}', '{}', {})",
-            runtime.node_id, config_name, entrypoint, initial_handle
+            "create_node('{}': '{}'.'{}')",
+            node_name, config_name, entrypoint
         );
         match self {
-            Configuration::LogNode => {
-                Box::new(logger::LogNode::new(config_name, runtime, initial_handle))
-            }
+            Configuration::LogNode => Some(Box::new(logger::LogNode::new(node_name))),
             Configuration::GrpcServerNode {
                 address,
                 tls_identity,
-            } => Box::new(grpc::server::GrpcServerNode::new(
-                config_name,
-                runtime,
+            } => Some(Box::new(grpc::server::GrpcServerNode::new(
+                node_name,
                 *address,
                 tls_identity.clone(),
-                initial_handle,
-            )),
-            Configuration::WasmNode { module } => Box::new(wasm::WasmNode::new(
-                config_name,
-                runtime,
-                module.clone(),
-                entrypoint,
-                initial_handle,
-            )),
-            Configuration::External => Box::new(external::PseudoNode::new(
-                config_name,
-                runtime,
-                initial_handle,
-            )),
+            ))),
+            Configuration::WasmNode { module } => {
+                match wasm::WasmNode::new(node_name, module.clone(), entrypoint) {
```

Comment:
(alternative)

```rust
wasm::WasmNode::new(node_name, module.clone(), entrypoint)
  .map(|node| Box::new(node))
```

Reply:
I tried that but it doesn't work (some kind of failure to convert between concrete type `wasm::WasmNode` and trait type`dyn node::Node`).